pkg/runtime/reconciler.go

```diff
@@ -194,10 +194,20 @@ func (r *resourceReconciler) reconcile(
 	res acktypes.AWSResource,
 ) (acktypes.AWSResource, error) {
 	if res.IsBeingDeleted() {
-		// Resolve references before deleting the resource.
-		// Ignore any errors while resolving the references
-		res, _ = rm.ResolveReferences(ctx, r.apiReader, res)
-		return r.deleteResource(ctx, rm, res)
+		// Determine whether we should retain or delete the resource
+		if r.getDeletionPolicy(res) == ackv1alpha1.DeletionPolicyDelete {
+			// Resolve references before deleting the resource.
+			// Ignore any errors while resolving the references
+			res, _ = rm.ResolveReferences(ctx, r.apiReader, res)
```

Comment:
QQ: Any reason why we need to resolve the references here?

Reply:
I didn't modify this code, here, I just added a conditional around it. But I believe we need to resolve references so that we have all the necessary fields to call `sdkDelete` if they require those fields.